pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -773,6 +773,19 @@ def test_aggregate_mixed_types():
     tm.assert_frame_equal(result, expected)
 
 
+def test_aggregate_categorical_lost_index():
+    # GH: 28641
+    result = (
+        pd.DataFrame(
+            {"A": [1997], "B": pd.Series(["b"], dtype="category").cat.as_ordered()}
+        )
+        .groupby("A")
+        .agg({"B": "min"})
+    )
```

Comment:
can you split this up into several lines, so that if it fails it is clear which step failed

Reply:
Done